### PR TITLE
Properly follow the spec in WebSocket::Constructor

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16099)

### CODE REVIEW

File: tests/wpt/metadata/websockets/cookies/007.html.ini [hunk 1]
```diff
@@ -1,5 +1,8 @@
 [007.html]
   type: testharness
+  [WebSockets: when to process set-cookie fields in ws response]
+    expected: FAIL
```
Comment (jdm) [inline tests/wpt/metadata/websockets/cookies/007.html.ini]: I filed https://github.com/w3c/web-platform-tests/issues/5213 about this.

**Comment by jdm** [comment]
@bors-servo: r+
Nice improvements!

**Comment by nox** [comment]
@bors-servo r=jdm

The failure was #13480, and I amended the PR to include a reference to https://github.com/w3c/web-platform-tests/issues/5213 in the corresponding expectation.

**Pull request merged**